src/figuresClasses.ts

```diff
@@ -1,19 +1,82 @@
+enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Colors = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shapes,
+  color: Colors,
+  getArea(): number,
+}
 
+function validSides(...sides: number[]): void {
+  if (sides.some((side) => side <= 0)) {
+    throw new Error('Each side value should be bigger than 0!');
+  }
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = Shapes.Triangle;
 
-}
+  constructor(
+    public color: Colors,
+    public a: number = 0,
+    public b: number = 0,
+    public c: number = 0,
+  ) {
+    validSides(a, b, c);
+
+    if (a + b <= c || a + c <= b || c + b <= a) {
+      throw new Error('Invalid side values!');
```

Comment:
I think here you check if is it possible to build a triangle with these sides. Error message can describe this better.